src/pages/index.js

```diff
@@ -4,78 +4,224 @@ import FormValidator from "../components/FormValidator.js";
+  })
+  .catch((error) => {
+    console.log(error);
+  });
+
+// Preview image
+function handleCardClick(name, link) {
+  previewImageModal.open(name, link);
+}
+
+// Add card function
+function handleAddCardSubmit(values) {
+  const name = values.title;
+  const link = values.link;
+    
+  // Set loading state
+  addCardPopup.setLoading(true);
+
+  // Call API method to add a new card
+  api.addNewCard(name, link)
+    .then((newCard) => {
+      // Render the newly created card
+      const cardElement = renderCard(newCard);
+
+      addCardPopup.setLoading(false);
+      addCardPopup.close();
+
+      // Ensure cardSection is accessible and addItem method is defined
+      if (cardSection && cardSection.addItem) {
+        cardSection.addItem(cardElement);
+      } else {
+        console.error('cardSection or addItem method is undefined.');
+      }
+      
+    })
+    .catch((error) => {
+      console.error('Error adding new card:', error);
+      // Reset loading state in case of an error
+    });
+}
+  
+// Update avatar function
+function handleAvatarUpdate(formData) {
+  updateAvatarPopup.setLoading(true);
+
+  // Get the link value from the input field
+  // const linkInput = document.getElementById('new-avatar-link-input');
+  // const link = linkInput.value;
+  const avatarUrl = formData.link;
+
+  // Ensure api.updateAvatar exists
+  if (typeof api.updateAvatar !== 'function') {
+    console.error('updateAvatar function is not defined in the API');
+    // updateAvatarPopup.setLoading(false); // Reset loading state in case of error
+    return;
+  }
+
+  // Call the API to update the avatar
+  api.updateAvatar(avatarUrl)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+      updateAvatarPopup.setLoading(false);
+      updateAvatarPopup.close(); // Ensure updateAvatarPopup is the correct reference
+    })
+    .catch((error) => {
+      console.error(error);
+      updateAvatarPopup.setLoading(false); // Reset loading state in case of error
+    });
+}
+
+// Delete card function
+function handleDeleteCard(card) {
+  confirmModal.open();
+
+  confirmModal.setSubmitAction(() => {
+    api.deleteCard(card.id)
+    .then(() => {
+      card.deleteCard();
+      confirmModal.close();
+    })
+    .catch((error) => {
+      console.log(error);
+    })
+  })
+}
+
+// Toggle like/dislike on card function
+function handleLikeCard(card) {
+  return api.toggleLikeCard(card.getId(), !card.getIsLiked())
+    .then((updatedCardData) => {
+      card.updateLikeValue(updatedCardData.isLiked);
+      return updatedCardData; // Assuming the API returns the updated card data
+    })
+    .catch((error) => {
+      console.log(error);
+      throw error; // Re-throw the error to be handled by the caller
+      
+    });
+}
+
+// Profile edit function
+function handleProfileEditSubmit(values) {
+  // Get the new profile data from the form inputs
+  const name = profileTitleInput.value;
+  const about = profileDescriptionInput.value;
```

Comment:
Only method `_getInputValues` collects all the data of the inputs **inside** `PopupWithForm` and passes it to the call of the `submit handler`. So you need to use the collected data rather than the inputs.

**Please, correct it everywhere**